Limits using isr

equinox0815 opened the pull request:
this should make limits handling quite a bit safer...

equinox0815 committed:
make eventqeue handling safe-ish

equinox0815 committed:
moving limits sliding window to isr

equinox0815 commented:
This is only compile tested yet!!

nicoonoclaste committed:
stepper: Avoid silly if/return dance

nicoonoclaste reviewed (approved):
Looks good to me, implements what we discussed.
I will merge once I reflash & test.

nicoonoclaste commented:
@equinox0815 Sorry for the delay, was kinda dead...

equinox0815 committed:
reduce ADC ringbuffer length to 4

This way the sum will react a little bit faster to changes and looking
at the generated code the calcualtion will be faster and therefore the ISR
will return faster.

equinox0815 committed:
fix limits thresholds

equinox0815 committed:
fix limits (enabling interrupt)

equinox0815 committed:
no extra steps after limits are reached
